### ISSUE TITLE
GLPK 4.65 causing test failures

### ISSUE THREAD

**Issue by blnicho** [Pyomo/pyomo#382]
It looks like GLPK on conda-forge was updated recently (4 days ago) from 4.63 to 4.65 and this update caused some of the GDP and bilevel book tests to start failing. We should check the solution for these tests and maybe update the `travis.yml` file to install a specific version of GLPK.

```
======================================================================
FAIL: baseline test: test_bilevel_ch_mip1a (test_book_examples.Test_bilevel_ch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 444, in <lambda>
    func = lambda self,c1=cwd,c2=cmd,c3=os.path.abspath(outfile),c4=baseline,c5=filter,c6=cmdfile,c7=tolerance,c8=exact,c9=forceskip: _run_cmd_baseline_test(self,cwd=c1,cmd=c2,outfile=c3,baseline=c4,filter=c5,cmdfile=c6,tolerance=c7,exact=c8,forceskip=c9)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 221, in _run_cmd_baseline_test
    outfile, baseline, filter=filter, tolerance=tolerance)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 363, in assertFileEqualsBaseline
    baseline + "\nDiffs:\n" + diffs)
AssertionError: Unexpected output difference at line 49:
   testfile=/home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/bilevel-ch/mip1a.out
   baseline=/home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/bilevel-ch/mip1a.txt
Diffs:
--- /home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/bilevel-ch/mip1a.txt

+++ /home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/bilevel-ch/mip1a.out

@@ -1,6 +1,6 @@

 Objective:
 o:
-Value: -21
+Value: -12.0
 sub.o:
-Value: 6
+Value: 4.0
 Variable:


======================================================================
FAIL: baseline test: test_gdp_ch_scont (test_book_examples.Test_gdp_ch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 444, in <lambda>
    func = lambda self,c1=cwd,c2=cmd,c3=os.path.abspath(outfile),c4=baseline,c5=filter,c6=cmdfile,c7=tolerance,c8=exact,c9=forceskip: _run_cmd_baseline_test(self,cwd=c1,cmd=c2,outfile=c3,baseline=c4,filter=c5,cmdfile=c6,tolerance=c7,exact=c8,forceskip=c9)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 221, in _run_cmd_baseline_test
    outfile, baseline, filter=filter, tolerance=tolerance)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 363, in assertFileEqualsBaseline
    baseline + "\nDiffs:\n" + diffs)
AssertionError: Unexpected output difference at line 63:
   testfile=/home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont.out
   baseline=/home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont.txt
Diffs:
--- /home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont.txt

+++ /home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont.out

@@ -1,4 +1,4 @@

 Value: 2
 x[2]:
-Value: 5
+Value: 6
 Constraint: No values


======================================================================
FAIL: baseline test: test_gdp_ch_scont2 (test_book_examples.Test_gdp_ch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 444, in <lambda>
    func = lambda self,c1=cwd,c2=cmd,c3=os.path.abspath(outfile),c4=baseline,c5=filter,c6=cmdfile,c7=tolerance,c8=exact,c9=forceskip: _run_cmd_baseline_test(self,cwd=c1,cmd=c2,outfile=c3,baseline=c4,filter=c5,cmdfile=c6,tolerance=c7,exact=c8,forceskip=c9)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 221, in _run_cmd_baseline_test
    outfile, baseline, filter=filter, tolerance=tolerance)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 363, in assertFileEqualsBaseline
    baseline + "\nDiffs:\n" + diffs)
AssertionError: Unexpected output difference at line 63:
   testfile=/home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont2.out
   baseline=/home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont2.txt
Diffs:
--- /home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont2.txt

+++ /home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont2.out

@@ -1,4 +1,4 @@

 Value: 2
 x[2]:
-Value: 5
+Value: 6
 Constraint: No values


======================================================================
FAIL: baseline test: test_gdp_ch_scont_script (test_book_examples.Test_gdp_ch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 444, in <lambda>
    func = lambda self,c1=cwd,c2=cmd,c3=os.path.abspath(outfile),c4=baseline,c5=filter,c6=cmdfile,c7=tolerance,c8=exact,c9=forceskip: _run_cmd_baseline_test(self,cwd=c1,cmd=c2,outfile=c3,baseline=c4,filter=c5,cmdfile=c6,tolerance=c7,exact=c8,forceskip=c9)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 221, in _run_cmd_baseline_test
    outfile, baseline, filter=filter, tolerance=tolerance)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/pyutilib/th/pyunit.py", line 363, in assertFileEqualsBaseline
    baseline + "\nDiffs:\n" + diffs)
AssertionError: Unexpected output difference at line 31:
   testfile=/home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont_script.out
   baseline=/home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont_script.txt
Diffs:
--- /home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont_script.txt

+++ /home/travis/build/Pyomo/pyomo/examples/doc/pyomobook/gdp-ch/scont_script.out

@@ -1,6 +1,6 @@

 0 :     0 :   2.0 :    20 : False : False :  Reals
 1 :     0 :   0.0 :    20 : False : False :  Reals
-2 :     0 :   5.0 :    20 : False : False :  Reals
+2 :     0 :   6.0 :    20 : False : False :  Reals
 
 Objectives:
 o : Size=1, Index=None, Active=True
```

**Comment by blnicho** [comment]
Here is the forum post that leads me to believe that GLPK is the problem http://lists.gnu.org/archive/html/bug-glpk/2018-03/msg00003.html

**Comment by blnicho** [comment]
Fixed by #395
